### ISSUE TITLE
[MPS] LayerNorm with fp32 weight/bias and fp16/bf16 input reinterprets the affine params; backward aborts

### ISSUE THREAD

**Issue by bogatyy** [pytorch/pytorch#190054]
### 🐛 Describe the bug

`layer_norm` on MPS selects its Metal kernel from the input dtype alone and
binds `weight`/`bias` buffers raw at that dtype. With an fp16 or bf16 input
and fp32 affine params, the fp32 bits are reinterpreted as fp16/bf16 and the output is
garbage. CPU (and CUDA) support this combination.

```python
import torch

torch.manual_seed(0)
for in_dtype in (torch.float16, torch.bfloat16):
    x = torch.randn(4, 8, dtype=in_dtype)
    w = torch.randn(8, dtype=torch.float32)
    b = torch.randn(8, dtype=torch.float32)
    ref = torch.nn.functional.layer_norm(x, (8,), w, b)
    out = torch.nn.functional.layer_norm(x.to("mps"), (8,), w.to("mps"), b.to("mps"))
    print(in_dtype, "max err:", (out.cpu().float() - ref.float()).abs().max().item())
```

Output on a source build of current main (2.14.0a0+git2662808); also
reproduces on 2.13.0 stable:

```text
torch.float16 max err: inf
torch.bfloat16 max err: 1.204268070216818e+31
```

Backward is worse: with `requires_grad` on the same mixed-dtype inputs,
MPSGraph rejects the fp16-times-fp32 multiply inside the backward graph and
the MLIR verifier assertion **aborts the process**:

```text
error: 'mps.multiply' op requires the same element type for all operands and results
MPSGraphExecutable.mm:1484: failed assertion `original module failed verification'
```

The forward kernels declare input, affine, and output buffers with one common
dtype:
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/kernels/LayerNorm.metal#L251-L293
and the host binds gamma/beta without a dtype check. The batch_norm paths in
the same file already handle this with `castMPSTensor`
(https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/Normalization.mm#L306-L307);
the layer_norm paths never adopted it.

### Versions

PyTorch 2.14.0a0+git2662808 (source build of main) and 2.13.0; macOS 26.5.2 (arm64), Apple M5 Max, Python 3.12.13.

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20
